lib/Service/RoomService.php

```diff
@@ -0,0 +1,86 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Service;
+
+
+use OCA\Talk\Exceptions\ParticipantNotFoundException;
+use OCA\Talk\Exceptions\RoomNotFoundException;
+use OCA\Talk\Manager;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCP\IUser;
+
+class RoomService {
+
+	/** @var Manager */
+	protected $manager;
+
+	public function __construct(Manager $manager) {
+		$this->manager = $manager;
+	}
+
+	public function createOneToOneConversation(IUser $actor, IUser $targetUser): Room {
+		if ($actor->getUID() === $targetUser->getUID()) {
+			throw new \InvalidArgumentException('invalid_invitee');
+		}
+
+		try {
+			// If room exists: Reuse that one, otherwise create a new one.
+			$room = $this->manager->getOne2OneRoom($actor->getUID(), $targetUser->getUID());
+			$room->ensureOneToOneRoomIsFilled();
+		} catch (RoomNotFoundException $e) {
+			$room = $this->manager->createRoom(Room::ONE_TO_ONE_CALL);
+			$room->addUsers([
+				'userId' => $actor->getUID(),
+				'participantType' => Participant::OWNER,
+			], [
+				'userId' => $targetUser->getUID(),
+				'participantType' => Participant::OWNER,
+			]);
+		}
+
+		return $room;
+	}
+
+	public function createConversation(int $type, string $name, ?IUser $owner = null, string $objectType = '', string $objectId = ''): Room {
+		$name = trim($name);
+		if ($name === '' || isset($name[255])) {
+			throw new \InvalidArgumentException('name');
+		}
+
+		$room = $this->manager->createRoom($type, $name, $objectType, $objectId);
+
+		if ($owner instanceof IUser) {
+			$room->addUsers([
+				'userId' => $owner->getUID(),
+				'participantType' => Participant::OWNER,
+			]);
+		}
+
+		return $room;
+	}
+
+	public function prepareConversationName(string $objectName): string {
+		return trim(mb_substr(trim($objectName), 0, 64));
```

Comment:
> I'm not sure whether silently truncating the room name is a good idea? I feel like this should be an error.

Since we sometimes automatically generate the name e.g. for file conversations or video verification, as well as when you select a group or circle, automatic truncate is the best idea I guess.

For manual name setting an error is better I agree, which is why `createEmptyRoom` doesn't prepare the name and instead returns a 400 as before.